ndcube/asdf/converters/extracoords_converter.py

```diff
@@ -0,0 +1,28 @@
+from asdf.extension import Converter
+
+
+class ExtraCoordsConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/extra_coords/extra_coords/extracoords-*"]
+    types = ["ndcube.extra_coords.extra_coords.ExtraCoords"]
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from ndcube.extra_coords.extra_coords import ExtraCoords
+        extra_coords = ExtraCoords()
+        extra_coords._wcs = node.get("wcs")
+        extra_coords._mapping = node.get("mapping")
+        extra_coords._lookup_tables = node.get("lookup_tables", [])
+        extra_coords._dropped_tables = node.get("dropped_tables")
+        extra_coords._ndcube = node.get("ndcube")
```

Comment:
Theoretically yes but I believe the converter will need to be updated.
https://asdf.readthedocs.io/en/latest/asdf/extending/converters.html#reference-cycles
That seems like a good test case.

Reply:
@braingram, for the current implementation I believe it is still not storing the `NDCube` object twice, right?
And also since the current `from_yaml_tree` doesn't have `yield`, it can still deserialize the extra_coords object, that is, when the extra_coords converter is called the extra_coords.ndcube shows `_PendingValue` and when it gets picked up in the NDCube converter and the `ndcube.extra_coords` is initialized the reference to the ndcube is properly mapped in the `ndcube.extra_coords._ndcube`.

Reply:
Thanks for testing this out. This looks to be working because `__set__` overwrites `_ndcube` when `_extra_coords` is assigned:
https://github.com/sunpy/ndcube/blob/09a83f14c704e1428bc931a3e02d4228a92743a6/ndcube/ndcube.py#L321
I am not sure what's going on with `__set__`. @Cadair is there ever an instance where:

- the `extra_coords` attached to a `NDCube` will reference a different `NDCube`?
- `extra_coords` will exist without a `NDCube`?

Reply:
>    the `extra_coords` attached to a `NDCube` will reference a different `NDCube`?

No, that should be explicitly prohibited by the descriptor.

>    `extra_coords` will exist without a `NDCube`?

I think this might be possible yes.

Reply:
> > `extra_coords` will exist without a `NDCube`?
> 
> I think this might be possible yes.

Is there a minimal example that would show this? Perhaps it can be used for a test case for the converter.

Reply:
I think you can just instantiate one standalone and then add some coordinates to it with the same API as you can if it's attached to the cube.

Reply:
turns out you can't: https://github.com/sunpy/ndcube/issues/753